pkg/controllers/provisioning/scheduling/requirements.go

```diff
@@ -74,22 +65,92 @@ func NewPodRequirements(pod *v1.Pod) Requirements {
 	return NewRequirements(requirements...)
 }
 
-// Add function returns a new Requirements object with new requirements inserted.
-func (r Requirements) Add(requirements ...v1.NodeSelectorRequirement) Requirements {
-	// Deep copy to avoid mutating existing requirements
-	r = *r.DeepCopy()
-	// This fail-safe measurement can be removed later when we implement test webhook.
-	if r.requirements == nil {
-		r.requirements = map[string]sets.Set{}
+func InstanceTypeRequirements(instanceTypes []cloudprovider.InstanceType) Requirements {
```

Comment:
Thoughts on keeping this in cloudprovider but having it return []v1.NodeSelectorRequirement? I really liked keeping the karpenter scheduler decoupled. You can convert it to a scheduling.Requirements in controllers/provisioner.go